### PR TITLE
vagrant cluster startup script: from zero to hero

### LINKED ISSUE #26: Post k8s deployment operations
Result of https://github.com/agalitsyn/yagoda/issues/1 is `vagrant up` which gives us a k8s cluster. It's not enough, we need:
- [ ] Ensure cluster is up and operational via http://kubernetes.io/docs/getting-started-guides/docker-multinode/testing/ 
- [ ] Deploy monitoring https://github.com/agalitsyn/yagoda/issues/13
- [ ] Deploy docker-registry https://github.com/agalitsyn/yagoda/issues/5 
- [ ] Build cassandra image and push it to registry
- [ ] Label nodes with cassandra tag

### PR DESCRIPTION
**Pull request by alexey-medvedchikov**
Depends on pull #35 pull #36 pull #37 pull #38 pull #39 

resolves #26

### CODE REVIEW

File: tools/k8s-vagrant.sh [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	vagrant_up
+	wait_for_k8s
+	k8s_cluster_check
+	label_nodes
+	deploy_registry
+	deploy_prometheus
+	build_and_push_cassandra
+}
+
+main $*
```
Comment (agalitsyn) [inline tools/k8s-vagrant.sh:82]: Not necessary to pass args, main function don't have any

File: tools/k8s-vagrant.sh [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+function k8s_cluster_check() {
+	kubectl run nginx-test --image=nginx --port=80
+	local pod=$(kubectl get pods -l run=nginx-test -o jsonpath='{ .items[0].metadata.name }')
+	echo "Got pod $pod"
+	kubectl expose pod $pod --target-port=80 --name=nginx-test --type=LoadBalancer
+	local host=$(kubectl get nodes \
+		-o jsonpath='{ .items[0].status.addresses[?(@.type == "InternalIP")].address }')
+	local nodeport=$(kubectl get service nginx-test -o jsonpath='{ .spec.ports[0].nodePort }')
+	echo -n "Awaiting http://$host:$nodeport "
+	while ! wget -qO /dev/null --tries 1 --timeout 1 http://$host:$nodeport ; do
```
Comment (agalitsyn) [inline tools/k8s-vagrant.sh]: I guess better to not use infinite loop. If it fails for example 30 times (1s sleep = 30 seconds) so we should terminate with failure report.

smth like that

``` sh
function wait-for-http() {
    local usage="wait_for_endpoint <endpoint> [poll_interval] [retries]"
    local endpoint=${1:?$usage}
    local poll_interval=${2:-1}
    local attempts=${3:-10}

    attempt=1
    until $(curl --output /dev/null --silent --head --fail $endpoint); do
        echo "Endpoint \"$endpoint\" is not available. Attempt $attempt of $attempts."

        if [[ $attempt -eq $attempts ]]; then
            die "all attempts were failed"
        fi

        sleep $poll_interval
        ((attempt++))
    done
}
```

File: tools/k8s-vagrant.sh [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+function vagrant_up() {
+	pushd "$SCRIPT_DIR/../vagrant/"
+	vagrant up
+	popd
+}
+
+function wait_for_k8s() {
+	local cluster_status="$SCRIPT_DIR/../vagrant/scripts/cluster_status.sh"
+	echo -n 'Waiting for k8s'
+	while ! bash $cluster_status >/dev/null 2>&1; do
```
Comment (agalitsyn) [inline tools/k8s-vagrant.sh]: Maybe here to, some fixed amount of attempts from my point of view is better

**Comment by agalitsyn** [comment]
Also I suggest to move script from `tools` folder to project root. Tools is kinda dev/test tools, but this script is for end-user.

**Comment by alexey-medvedchikov** [comment]
@agalitsyn Fresh berries arrived

File: tools/k8s-vagrant.sh (hunk 1 above)
Comment (alexey-medvedchikov) [inline tools/k8s-vagrant.sh:82]: I often forget to do that, so if it will... I'll spent couple minutes to figure out what's wrong :)

**Comment by agalitsyn** [comment]
Awesome :+1:

**Pull request merged**